students/carlos_novoa/lesson08/circle_class.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import math
+
+"""
+Lesson8 - Circle Class
+"""
+
+
+class Circle(object):
+    version = '0.1'
+
+    def __init__(self, radius=None, diameter=None):
+
+        if radius is None and diameter is None:
+            raise ValueError()
+
+        if radius is not None:
+            try:
+                if radius < 0:
+                    raise ValueError()
+            except ValueError:
+                print('Radius must be positive.')
+                return
+            else:
+                self._radius = radius
+                self._diameter = radius * 2
```

Comment:
You don't need to calculate this here, since you are doing it below in the diameter `@property`. The way properties work, is that they run each time the attribute that they define is accessed on the object instance. So every time that your program accesses `circle.diameter`, the diameter will be calculated in your `@property` based on the current radius value. This is why the `@property` decorator is often used for values that are calculated from the values of other attributes.